PaulasWeatherApp/PaulasWeatherApp/ViewController/CurrentWeatherViewController.swift

```diff
@@ -8,11 +8,108 @@
 import UIKit
 
 class CurrentWeatherViewController: UIViewController {
-   
+        
+    private let tempLabel = UILabel()
+    private let sunriseLabel = UILabel()
+    private let sunsetLabel = UILabel()
+    
+    private var sunsetString = NSMutableAttributedString()
+    private var sunriseString = NSMutableAttributedString()
+    
+    let activityIndicator = UIActivityIndicatorView(style: .large)
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view.
+        setActivityIndicator()
+        
+        sunsetString = createString(for: "sunset")
+        sunriseString = createString(for: "sunrise")
+        
+        setTempLabel()
+        setSunriseLabel()
+        setSunsetLabel()
         view.backgroundColor = .systemRed
     }
     
+    private func setActivityIndicator() {
+        activityIndicator.center = view.center
+        activityIndicator.hidesWhenStopped = true
+        view.addSubview(activityIndicator)
+        activityIndicator.startAnimating()
```

Comment:
The activity indicator should be animated when there's something that is loading, like a call in progress. It shouldn't start animating directly when set. A better approach would be to have a method, something like _loadingData()_ where you can animate the activity indicator